src/main/java/org/mule/tools/devkit/sonar/utils/PomUtils.java

```diff
@@ -54,20 +72,49 @@ public static boolean isDevKitConnector(MavenProject mavenProject) {
         return parent != null && parent.getGroupId().equals(DEVKIT_GROUP_ID) && (parent.getArtifactId().equals(DEVKIT_ARTIFACT_ID) || parent.getArtifactId().equals(CERTIFIED_DEVKIT_ARTIFACT_ID));
     }
 
+    public static VersionUtils getCurrentDevkitVersion(String devkitVersion) {
+        return new VersionUtils(devkitVersion);
+    }
+
+    public static VersionUtils getLatestDevkitVersion() {
+        VersionUtils latestVersion = getCurrentDevkitVersion("0.0.0");
+        try (InputStream xml = new URL("https://repository.mulesoft.org/nexus/content/repositories/releases/org/mule/tools/devkit/mule-devkit-parent/maven-metadata.xml").openStream()) {
+            Document doc = DocumentBuilderFactory.newInstance().newDocumentBuilder().parse(xml);
+            doc.getDocumentElement().normalize();
+            NodeList versions = doc.getElementsByTagName("version");
+            latestVersion = getCurrentDevkitVersion(versions.item(0).getFirstChild().getTextContent());
+            for (int i = 1; i < versions.getLength(); i++) {
+                Node tag = versions.item(i);
+                String currentValue = tag.getNodeType() == ELEMENT_NODE ? tag.getFirstChild().getTextContent() : EMPTY;
+                // Ignore revisions (e.g. 3.7.0-M1) and Mule 4.x.x versions that refer to the new SDK
+                if (isRevision(currentValue) && isNotEmpty(currentValue) && currentValue.indexOf('4') != 0) {
```

Comment:
Martin, you mean because of the indexOf('4')? In that case, it only evaluates whether that 4 is the major version, in which case it would be a migration tool.